src/crypto/tls/ech.go

```diff
@@ -1008,3 +1009,50 @@ func (c *Config) echCanAccept() bool {
 		c.ServerECHProvider != nil &&
 		c.maxSupportedVersion() >= VersionTLS13
 }
+
+var (
+	// The list of compressed extensions used for the "ech_outer_extensions"
+	// extension.
+	echOuterExtensions []uint16
+)
+
+func init() {
+	// EncodedClientHelloInner is constructed from ClientHelloInner by removing
+	// extensions that are also used in the ClientHelloOuter.
+	//
+	// NOTE(cjpatton): It would be nice to incorporate more extensions, but
+	// "key_share" is the last extension to appear in the ClientHello before
+	// "pre_shared_key". As a result, the only contiguous sequence of outer
+	// extensions that contains "key_share" is "key_share" itself. Note that
+	// we cannot change the order of extensions in the ClientHello, as the
+	// unit tests expect "key_share" to be the second to last extension.
+	echOuterExtensions = []uint16{extensionKeyShare}
```

Comment:
Why is this stuff here and not in test_ech.go?

Reply:
Hmmm, well I suppose we could have two `init()` functions, one here and one in test_ech.go, where the latter overwrites the value of `echOuterExtensions` set here. But I think it's more readable to have the logic for setting this value all in one function.

Reply:
`init` is rarely used in the library outside tests. Does this need to be an `init`?

Reply:
Nope! I replaced the `init()` with a function `echOuterExtensions()` that generates this list.